src/js/classes/BooleanValidator.ts

```diff
@@ -0,0 +1,60 @@
+import PropTypes from 'prop-types';
+import BaseValidator from './BaseValidator';
+import {Dependent, ErrorFunction, InputType, MutationFunction, ValidationResult, Value} from '../types';
+
+/**
+ * A validator that allows you to validate boolean fields.
+ * @extends BaseValidator
+ */
+class BooleanValidator extends BaseValidator<boolean | string> {
+    public defaultInputType: InputType = 'checkbox';
+    protected propType = PropTypes.bool;
+    private realBool: boolean;
+
+    /**
+     * Validate a boolean field
+     * @param {Boolean} [defaultValue]
+     * @param {String} [defaultErrorMsg]
+     * @param {MutationFunction} [mutationFunc]
+     * @param {ErrorFunction} [onError]
+     * @param {Dependent} [dependent]
+     * @param {Boolean} [realBool]
+     */
+    constructor(
+        defaultValue: boolean = false,
+        defaultErrorMsg: string = 'This field has to be a boolean',
+        mutationFunc?: MutationFunction,
+        onError?: ErrorFunction,
+        dependent?: Dependent,
+        realBool = false,
```

Comment:
I think since this will only be released as a breaking 2.0 release, we should set realBool to true by default since most use cases will need the real boolean